### ISSUE TITLE
Add addBuildChainCustomizer to QuarkusDevModeTest

### ISSUE THREAD

**Issue by ia3andy** [quarkusio/quarkus#40121]
### Description

Currently it's only possible in `QuarkusUnitTest`.

The example of usage is when a SPI BuildItem needs to be tested in devmode, it's currently not possible.

**Comment by ia3andy** [comment]
It would be needed for https://github.com/quarkusio/quarkus/pull/39891/

**Comment by geoand** [comment]
I am assuming that just copying the relevant code from `QuarkusUnitTest` to `QuarkusDevModeTest` would work

**Comment by mcruzdev** [comment]
@ia3andy for me, it is not as simple as I expected 😆, then I will do our workaround first

**Comment by ia3andy** [comment]
ok, don't hesitate to tell if this is very complex, either @geoand will go on, or else it means it's not possible 🤣

**Comment by mcruzdev** [comment]
yes, this is complex, I can spend time with this one, but I prefer to finish the another one.

**Comment by geoand** [comment]
Okay, now I see that this indeed far more complicated than I initially thought.  It's likely doable, but almost certainly not worth it under any circumstance.
